src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package pl.byczazagroda.trackexpensesappbackend.service;
+
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.validation.annotation.Validated;
+import pl.byczazagroda.trackexpensesappbackend.dto.FinancialTransactionCategoryDTO;
+import pl.byczazagroda.trackexpensesappbackend.dto.FinancialTransactionCategoryDetailedDTO;
+import pl.byczazagroda.trackexpensesappbackend.exception.AppRuntimeException;
+import pl.byczazagroda.trackexpensesappbackend.exception.ErrorCode;
+import pl.byczazagroda.trackexpensesappbackend.mapper.FinancialTransactionCategoryModelMapper;
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransactionCategory;
+import pl.byczazagroda.trackexpensesappbackend.repository.FinancialTransactionCategoryRepository;
+import pl.byczazagroda.trackexpensesappbackend.repository.FinancialTransactionRepository;
+
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+import java.math.BigInteger;
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+@Validated
+public class FinancialTransactionCategoryServiceImpl implements FinancialTransactionCategoryService {
+
+    private final FinancialTransactionCategoryRepository financialTransactionCategoryRepository;
+    private final FinancialTransactionCategoryModelMapper financialTransactionCategoryModelMapper;
+    private final FinancialTransactionRepository financialTransactionRepository;
+
+    @Override
+    public List<FinancialTransactionCategoryDTO> getFinancialTransactionCategories() {
+        return financialTransactionCategoryRepository.findAll().stream()
+                .map(financialTransactionCategoryModelMapper::mapFinancialTransactionCategoryEntityToFinancialTransactionCategoryDTO)
+                .toList();
+    }
+
+    @Override
+    public FinancialTransactionCategoryDetailedDTO findById(@Min(1) @NotNull Long id) {
```

Comment:
Name of variables refactoring suggestion:
```java
@Override
public FinancialTransactionCategoryDetailedDTO findById(@Min(1) @NotNull Long id) {       
	FinancialTransactionCategory ftcEntity = financialTransactionCategoryRepository.findById(id)
			.orElseThrow(() -> new AppRuntimeException(ErrorCode.FTC001,
					String.format("Financial transaction category with id: %d not found", id)));
					
	FinancialTransactionCategoryDTO ftcDTO = financialTransactionCategoryModelMapper
			.mapFinancialTransactionCategoryEntityToFinancialTransactionCategoryDTO(ftcEntity);

	BigInteger counter = BigInteger.valueOf(financialTransactionRepository.countFinancialTransactionsByFinancialTransactionCategoryId(id));

  return new FinancialTransactionCategoryDetailedDTO(ftcDTO, counter);
}
```

Reply:
Thank you, I corrected as you suggested